modules/core/04-channel/keeper/keeper.go

```diff
@@ -623,17 +623,17 @@ func (k Keeper) HasInflightPackets(ctx sdk.Context, portID, channelID string) bo
 	return iterator.Valid()
 }
 
-// SetPruningSequenceEnd sets the channel's pruning sequence end to the store.
-func (k Keeper) SetPruningSequenceEnd(ctx sdk.Context, portID, channelID string, sequence uint64) {
+// SetRecvStartSequence sets the channel's recv start sequence to the store.
```

Comment:
should we add an extra note, with something like:

> NOTE: RecvStartSequence is only set for channels which have undergone an upgrade. This sequences acts as a checkpoint for pruning historical channel data 

could also add a note saying its used as the upper bound when pruning

Reply:
or maybe something specific like "This sequence indicates the first packet sequence which should be processed after an upgrade"

Reply:
I thought about this, but none of the other set functions for `nextSequenceRecv` `nextSequenceAck` indicate their purpose. I wasn't sure where we should be documenting their behaviour (presumably at least in the spec). I can add a comment in you'd like?

Reply:
https://github.com/cosmos/ibc-go/pull/5651/commits/b2a1f50b39029594c48c1189e4853e7b24d597f6